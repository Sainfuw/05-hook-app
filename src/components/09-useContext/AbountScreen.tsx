import React from "react";

export const AbountScreen = () => {
  return (
    <>
      <h1>AbountScreen</h1>
      <hr />
    </>
  );
};
