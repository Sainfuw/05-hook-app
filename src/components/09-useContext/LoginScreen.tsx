import React, { useContext } from "react";
import { UserContext } from "./UserContext";
// import { useHistory } from "react-router-dom";

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);
  // let history = useHistory();

  const handleLogin = () => {
    setUser({
      id: 12345,
      name: "Emilio Navarro",
      email: "sainfuw@aol.com",
    });
    // history.push("/home");
  };

  return (
    <>
      <h1>LoginScreen</h1>
      <hr />

      <button className="btn btn-outline-primary" onClick={handleLogin}>
        Login
      </button>
    </>
  );
};
