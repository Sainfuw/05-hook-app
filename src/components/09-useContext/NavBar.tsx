import React from "react";
import { NavLink } from "react-router-dom";

export const NavBar = () => {
  return (
    <nav className="navbar navbar-expand navbar-dark bg-dark mb-3">
      <NavLink className="navbar-brand" to="/">
        useContext
      </NavLink>
      <div className="" id="base">
        <ul className="navbar-nav me-auto mb-2 mb-lg-0">
          <li className="nav-item">
            <NavLink className="nav-link" exact activeClassName="active" to="/">
              Home
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink
              className="nav-link"
              activeClassName="active"
              exact
              to="/about"
            >
              About
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink
              className="nav-link"
              activeClassName="active"
              exact
              to="/login"
            >
              Login
            </NavLink>
          </li>
        </ul>
      </div>
    </nav>
  );
};
