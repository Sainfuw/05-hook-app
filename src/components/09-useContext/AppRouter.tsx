import React from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from "react-router-dom";
import { AbountScreen } from "./AbountScreen";
import { LoginScreen } from "./LoginScreen";
import { HomeScreen } from "./HomeScreen";
import { NavBar } from "./NavBar";

export const AppRouter = () => {
  return (
    <Router>
      <div className="">
        <NavBar />
        <div className="container">
          <Switch>
            <Route exact path="/about" component={AbountScreen} />
            <Route exact path="/login" component={LoginScreen} />
            <Route exact path="/" component={HomeScreen} />

            {/* Default route en caso de error */}
            {/* <Route component={HomeScreen} /> */}

            {/* En caso de no encontrar ruta correcta redirecciona a la raiz */}
            <Redirect to="/" />
          </Switch>
        </div>
      </div>
    </Router>
  );
};
