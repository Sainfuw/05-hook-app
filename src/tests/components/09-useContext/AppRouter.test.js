import "@testing-library/jest-dom";
import { mount } from "enzyme";
import { AppRouter } from "../../../components/09-useContext/AppRouter";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("Pruebas en AppRouter.test", () => {
  const user = {
    name: "Emilio",
    email: "sainfuw@aol.com",
  };

  const wrapper = mount(
    <UserContext.Provider
      value={{
        user,
      }}
    >
      <AppRouter />
    </UserContext.Provider>
  );

  test("debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
