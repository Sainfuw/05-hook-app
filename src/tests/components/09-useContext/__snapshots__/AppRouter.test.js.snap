// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter.test debe mostrarse correctamente 1`] = `
<div
  className=""
>
  <nav
    className="navbar navbar-expand navbar-dark bg-dark mb-3"
  >
    <a
      aria-current="page"
      className="navbar-brand active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      useContext
    </a>
    <div
      className=""
      id="base"
    >
      <ul
        className="navbar-nav me-auto mb-2 mb-lg-0"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="container"
  >
    Array [
      <h1>
        HomeScreen
      </h1>,
      <hr />,
      <pre>
        {
   "name": "Emilio",
   "email": "sainfuw@aol.com"
}
      </pre>,
    ]
  </div>
</div>
`;
